doc/tutorials/imgproc/shapedescriptors/bounding_rects_circles/bounding_rects_circles.markdown

```diff
@@ -22,6 +22,97 @@ This tutorial code's is shown lines below. You can also download it from
+    blur( src_gray, src_gray, Size(3,3) );
+    @endcode
+-# Create a window with header "Source" and display the source file in it.
+    @code{.cpp}
+    /// Create Window
+    char* source_window = "Source";
+    namedWindow( source_window, CV_WINDOW_AUTOSIZE );
+    imshow( source_window, src );
+    @endcode
+-# Create a trackbar on the source_window and assign a callback function to it
+   In general callback functions are used to react to some kind of signal, in our
+   case it's trackbar's state change.
+    @code{.cpp}
+    createTrackbar( " Threshold:", "Source", &thresh, max_thresh, thresh_callback );
+    @endcode
+
+The callback function `threshold` does all the interesting job.
+-# Writes to `threshold_output` the threshold of the grayscale picture, (you can check out about thresholding [here](http://docs.opencv.org/2.4/doc/tutorials/imgproc/threshold/threshold.html))
+    @code{.cpp}
+    /// Detect edges using Threshold
+    threshold( src_gray, threshold_output, thresh, 255, THRESH_BINARY );
+    @endcode
+-# Finds contours and saves them to the vectors `contour` and `hierarchy` ([more about findContours](http://docs.opencv.org/2.4/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html#void%20findContours(InputOutputArray%20image,%20OutputArrayOfArrays%20contours,%20OutputArray%20hierarchy,%20int%20mode,%20int%20method,%20Point%20offset))).
+    @code{.cpp}
+    findContours( threshold_output, contours, hierarchy, RETR_TREE, CHAIN_APPROX_SIMPLE, Point(0, 0) );
+    @endcode
+-# For every found contour we now apply approximation to polygons
+   with accuracy +-3 and stating that the curve must me closed [approxPolyDP](http://www.opencv.org.cn/opencvdoc/2.3.1/html/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html#approxpolydp).
+   After that we find a bounding rect for every polygon and save it to `boundRect` (more about [boundingRect](http://docs.opencv.org/2.4/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html?highlight=boundingrect#boundingrect)).
+   At last we find a minimum enclosing circle for every polygon and
+   save it to `center` and `radius` vectors (more about [minenclosingcircle](http://docs.opencv.org/2.4/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html?highlight=boundingrect#minenclosingcircle)).
+    @code{.cpp}
+      for( size_t i = 0; i < contours.size(); i++ )
+      {
+        approxPolyDP( Mat(contours[i]), contours_poly[i], 3, true );
+        boundRect[i] = boundingRect( Mat(contours_poly[i]) );
+        minEnclosingCircle( contours_poly[i], center[i], radius[i] );
+      }
+    @endcode
+
+We found everything we need, all we have to do is to draw.
+
+-# Create new Mat of unsigned 8-bit chars, filled with zeros (more about [Mat::zeros](http://docs.opencv.org/master/d3/d63/classcv_1_1Mat.html#a0b57b6a326c8876d944d188a46e0f556)).
+   It will contain all the drawings we are going to make (rects and circles).
+    @code{.cpp}
```

Comment:
Please use `@snippet` command to insert code snippets from existing source file. The documentation writing guide has [corresponding section](http://docs.opencv.org/master/d4/db1/tutorial_documentation.html#tutorial_documentation_commands_include) with more information about preparing source file and using this command.